pandas/tests/indexes/common.py

```diff
@@ -682,6 +682,21 @@ def test_hasnans_isnans(self):
                 self.assert_numpy_array_equal(idx._isnan, expected)
                 self.assertTrue(idx.hasnans)
 
+    def test_ensure_copied_data(self):
+        # GH12309
+        for name, index in compat.iteritems(self.indices):
+            if isinstance(index, (PeriodIndex, RangeIndex, MultiIndex)):
```

Comment:
so then these need specific tests (name this the same as this one and put in the appropriate place)